### ISSUE TITLE
SIM card contacts can be (and usually are) EVIL: use double confirmation

### ISSUE THREAD

**Issue by xandro0777** [SimpleMobileTools/Simple-Dialer#595]
Opening it here because it is the dialer app where it can cause greatest damage.

Seen https://github.com/SimpleMobileTools/Simple-Contacts/issues/707 - my issue is, many if not most SIM card vendors add quite a few evil pay service numbers to their freshly issued SIM cards. Even worse those may be undeletable or magically reappear after being deleted (seen that), rumors claim new pay service numbers being pushed OTA, SIM card numbers may (perhaps depending on Android version/vendor) overwrite phone contacts.

There may be some users that use SIM cads to migrate contacts with old trusted SIM cards but for many SIM card contacts is a never used feature, nuisance or outright threat. No shortage of reports of people having accidentally dialed on of those 0900 numbers inside their pocket. 

Is it possible to distinguish this two cases? If not... I would ask to implement some kind of double confirmation before dialing one of the contacts from a SIM card. Even Google (on my phone) asks me to review the SIM card contacts before adding them to the contact database.

Or maybe totally ignore SIM card contacts by default? After all.. there are specialized apps that can import/export and otherwise manage SIM card contacts, maybe this is simply out of scope for the simple mobile tools.

**Comment by tibbi** [comment]
no idea what you mean at all, you bought some SIM and it is not empty, but contains some numbers?

**Comment by xandro0777** [comment]
Yes. It contains plenty of preprogrammed contacts for $$$ services, many of them starting with "A" and not deletable by simple mobile tools.

**Comment by tibbi** [comment]
can you delete them in any other downloaded app?

**Comment by tibbi** [comment]
I never heard about such contacts, didnt even know it exists... I can keep it here but dont expect having it solved anything soon

**Comment by lefterisgar** [comment]
I can confirm what @xandro0777 says is true. In my country, most carriers include a dozen service numbers in each SIM card they give you. Some of these are free to use (if you need technical support) but there are a few premium-rate phone numbers as well. In my case, I've deleted them a few years ago and they didn't reappear.
